Project organization refactor

jooola opened the pull request:
Copy pasted the reasons for reference:

Some reasons on what changed in this PR :

- First we should avoid having naming conflicts with *Unix folders naming. The var folder is supposed to get application run-time data (such as logs/pictures/databases), not build/test output, the /build or /dist folder is more common for this purpose. The same goes for etc, which is supposed to handle software run-time configuration (such as webservers/php message broker configurations), not development tools configuration files.

- Having them at the root or the project gives a better visibility, on what tools is used to maintain this project and it is easier for newcomers/everyone to embrace the default. That being said, I can understand that having all those files in the root folders might seems like a mess, this is why I didn't used the fully default configuration file names in order to hide them on linux. I would prefer using the default names, but I though the main idea of the etc folder was to clean the root folder.

- You mentionned that it will not be a common app/project, but more like a library/package. I would argue that gives us even more freedom to move dev tools config files to the root folder since it will be installed using composer.

- Convention over configuration => development tools embed in IDE's use the default configuration paths to run there respective tasks, would be nice to embrace the default and avoid more configuration.

- Avoid unnecessary complexity: path need to go to parent folder first. (../)

I felt like hiding them using dotfiles was the in-between that might be accepted, but I suggest to use the default configuration names.

jooola commented:
@WoLfulus To make your life easier, review the PR per commit, I know I should have split this into many small PR's but commit history is pretty clean.

WoLfulus commented on .gitignore:
```diff
@@ -1,12 +1,98 @@
+## Directus custom .gitignore
+################################################################################
+
 /dist
```
I'd like to keep the .gitignore file clean and add stuff only when really needed needed.

jooola replied:
Ok, removed the big gitignore supercharge commit

WoLfulus commented on .gitignore:
```diff
@@ -1,12 +1,98 @@
+# IDE
 /.idea
 /.vscode
-/phpunit.xml
```
This should be removed anyways

WoLfulus commented on .phpunit.xml:
```diff
@@ -1,20 +1,26 @@
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/8.5/phpunit.xsd"
-         bootstrap="vendor/autoload.php"
+         xsi:noNamespaceSchemaLocation="./vendor/phpunit/phpunit/phpunit.xsd"
```
Nevermind my first comment on this line.

WoLfulus commented on composer.json:
```diff
@@ -52,8 +52,8 @@
     "test": "phpunit --colors=always --configuration=.phpunit.xml",
-    "test:with-coverage": "phpunit --colors=always --configuration=.phpunit.xml --log-junit=build/phpunit.xml --coverage-html=build/html --coverage-clover=build/coverage.xml",
-    "coverage:report": "coveralls build/coverage.xml",
+    "test:with-coverage": "phpunit --colors=always --configuration=.phpunit.xml",
```
Shouldn't we remove `--colors=always` from both as it's now specified in .phpunit.xml:7?

jooola replied:
I had troubles making the colors work when only specified in the phpunit.xml

I'll take a look at it.

jooola replied:
https://phpunit.readthedocs.io/en/9.0/configuration.html#the-colors-attribute

We need to force it using the CLI flag

jooola commented on composer.json:
```diff
@@ -6,15 +6,10 @@
     "directus",
     "laravel"
   ],
-  "authors": [
```
Just tried, it won't.

jooola commented:
@WoLfulus I fixed the issues you raised, but this PR needs another quick look before merging. Sorry I force pushed on the branch since I had to rebase on master that also has been force pushed, couldn't make something clean otherwise. I also squashed few commits.

jooola commented:
Also CI has not been respected so we cannot make it pass here. Needs another PR (formatting is broken and phpunit 9 complains about the PHP version)

WoLfulus commented:
Thank you!
